identity/webapp/src/app.ts

```diff
@@ -28,7 +29,79 @@ export async function createApp(): Promise<Koa> {
   app.use(apmErrorMiddleware);
 
   app.use(json({ pretty: process.env.NODE_ENV !== 'production' }));
-  app.use(logger());
+
+  // This custom logger query parameter values from the logs.
+  //
+  // This is to prevent PII from being written to the logs, in particular
+  // email addresses and other personal information being passed around
+  // as JWTs as part of the sign-up process.
+  //
+  // The default logger might log something like:
+  //
+  //      --> GET /account/registration?session_token=eyJhbGciOi...{some sort of jwt}
+  //
+  // where the JWT could be decoded to reveal personal information including
+  // email address and IP address.
+  //
+  // This logger will replace this with:
+  //
+  //      --> GET /account/registration?session_token=[redacted]
+  //
+  // It is deliberately un-picky about what it redacts, because we'd rather
+  // remove something innocent (e.g. ?refresh=true) than miss something sensitive.
+  //
+  // The goal is to keep PII about library users out of the logging cluster.
+  //
+  // See:
+  //
+  //    * Documentation for custom transporters
+  //      https://github.com/koajs/logger#use-custom-transporter
+  //    * Slack discussion about access tokens in logs
+  //      https://wellcome.slack.com/archives/CUA669WHH/p1656593455081159
+  //
+  app.use(
+    logger({
+      transporter: (_, args) => {
+        const [format, method, url, status, time, length] = args;
+
+        // Note: we use a deprecated API here because we're working with
+        // relative URLs, e.g. `/account`.
+        //
+        // The WHATWG URL API that the deprecation message suggests doesn't
+        // work for this use case; it wants absolute URLs.
+        const parsedUrl = parse(url);
+        const params = new URLSearchParams(parsedUrl.query);
```

Comment:
I did this because we already use the `deprecated` parse API elsewhere; I’m going to look at replacing that in a separate PR, then come back to this use.

Reply:
Okay, no I’m not, because this is waaaaay more complicated than I realised. 😭 

This particular API has been marked as "legacy" rather than "deprecated" precisely because so many people need to do stuff with relative URLs that aren't supported by the WHATWG spec: https://nodejs.org/dist/latest-v18.x/docs/api/url.html#legacy-url-api